python/pyspark/statcounter.py

```diff
@@ -53,12 +54,12 @@ def merge(self, value):
         return self
 
     # Merge another StatCounter into this one, adding up the internal statistics.
-    def mergeStats(self, other):
+    def mergeStats(self, other: "StatCounter") -> "StatCounter":
         if not isinstance(other, StatCounter):
             raise TypeError("Can only merge StatCounter but got %s" % type(other))
 
         if other is self:  # reference equality holds
-            self.merge(copy.deepcopy(other))  # Avoid overwriting fields in a weird order
+            self.mergeStats(copy.deepcopy(other))  # Avoid overwriting fields in a weird order
```

Comment:
Let's `type: ignore` handle this as a separate ticket. This has been here forever and was never reported, so there is good indication it is a piece of dead code, and we should cover this scenario with unit tests anyway.

Reply:
Should I just add # type: ignore like: `self.merge(copy.deepcopy(other))  # type: ignore` and create a new Jira ticket with UT for this issue? @zero323 , @ueshin , @HyukjinKwon 
Many thanks!

Reply:
Yeah, let's handle it in a separate PR.